bql/planner/tracer/trace.go

```diff
@@ -25,19 +25,32 @@ type Arguments struct {
 	Msgs []string
 }
 
+// event encapsulates a single tracing event.
 type event struct {
 	w          io.Writer
 	t          time.Time
 	tracerArgs func() *Arguments
 }
 
-var c chan *event
+// MessageTracer encapsulates the intrinsic verbosity of a given tracing message.
+type MessageTracer struct {
```

Comment:
MessageTracer doesn't seem to be used in other packages. Is this expected? If yes, maybe we should make it unexported (lowercase)

Reply:
Also I'm wondering if it's possible to avoid reading and writing a package-level variable. Does it make sense to have a constructor for the message tracer where we pass the "global verbosity"?

Reply:
About `MessageTracer`, I did indeed make it unexported (lowercase) when first writing this code. But, since the public/exported `tracer.V` method was returning a `tracer.messageTracer` unexported type (to have its public `.Trace` method used in other packages), I was getting a persistent warning here on `VSCode` at the line the `tracer.V` method was being defined (saying that it was a public method return a not-public type).

Given this, to solve this warning, I decided to make `MessageTracer` a exported type.

In this situation, should I keep it exported to resolve the warning or make it unexported as I did at the beginning and ignore this warning in the IDE?

Reply:
About your second question, the objective of the `MessageTracer` is just to encapsulate the `.Trace` method for a given message, keeping track of the intrinsic verbosity level of that message and comparing it to the global verbosity of the current tracer before sending it to the tracing output. Its "constructor" is just the `tracer.V` method at the moment, and having the "global tracer verbosity" passed inside of it would be out of its scope.

But I understand you when worrying about "reading and writing a package-level variable". We also think about this and for the future of the tracer we are planning to make it an object, to be created by a `.New` method, having inside of it this "global tracer verbosity" field, and having the correspondent methods the `tracer` package has at the moment. This way, we would not have reads and writes to a package-level variable anymore.

Reply:
1. Right, the warning makes sense, as you are accessing the Tracer method we should maintain as exported.

2.I agree. Probably better to have a New constructor for a generic Tracer struct and from that struct create a message tracer with a specific verbosity. Maybe leave a TODO?

Reply:
In the design doc we have here for the future of the tracer we already have a TODO there: an entire section dedicated for this future refactoring that will create this `.New()`, and make the tracer a struct with methods.

Thank you again Jose.